File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/query/expr/DynamicConstraintExpr.ts
module api.query.expr {

    export class DynamicConstraintExpr implements ConstraintExpr {
        private func: FunctionExpr;

        constructor(func: FunctionExpr) {
            this.func = func;
        }

        getFunction(): FunctionExpr {
            return this.func;
        }

        toString() {
            return this.func.toString();
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/query/expr/DynamicConstraintExpr.java
package com.enonic.xp.query.expr;

import com.google.common.annotations.Beta;

@Beta
public final class DynamicConstraintExpr
    implements ConstraintExpr
{
    private final FunctionExpr function;

    public DynamicConstraintExpr( final FunctionExpr function )
    {
        this.function = function;
    }

    public FunctionExpr getFunction()
    {
        return this.function;
    }

    @Override
    public String toString()
    {
        return this.function.toString();
    }
}
